tests/test_com_aux.py

```diff
@@ -183,9 +182,15 @@ def test_receive_message_exception(mocker, com_aux_init, caplog):
     )
 
 
-def test_receive_message_none(mocker, com_aux_init):
+@pytest.mark.parametrize("clear_buffer", [True, False])
+def test_receive_message_none(mocker, com_aux_init, clear_buffer):
     mocker.patch.object(com_aux_init, "wait_and_get_report", return_value=None)
-    recv = com_aux_init.receive_message(2)
+    mock_empty = mocker.patch.object(queue.Queue, "empty")
```

Comment:
Wow, NO!
You should create a test where you enqueue x message, call the function, enqueue a new message, and compare that what is received is the last one, and that the queue is than empty!

Reply:
Uhhh, I guess method empty from queue.Queue is working properly ^^

Reply:
Maybe, but you nevertheless, still not test the new feature. What if I do not want to delete the messages and want to somehow create a pattern analysis out of it?
Due to the merging conflicts, the automation is not running, but I'm sure codecov will flag you red here ^^